### PR TITLE
Fixes #16632: Add `testCases` property to a test suite source config 

### LINKED ISSUE #16632: Support test case filtering in test suite pipeline
**Is your feature request related to a problem? Please describe.**
Required for https://github.com/open-metadata/OpenMetadata/issues/15133.

**Describe the solution you'd like**
Add the option to filter a subset of test cases when running a test suite pipeline.

**Describe alternatives you've considered**
There is no option to run a subset of tests for a given table. Any test suite pipeline run runs all the tests defined.

### PR DESCRIPTION
**Pull request by sushi30**
### Describe your changes:

enable specifying a subset of test cases to run in a test suite pipeline.

no schema migration is required since the default is null.

Fixes https://github.com/open-metadata/OpenMetadata/issues/16632

adds the `testCases` property to [testSuitePipeline.json](https://github.com/open-metadata/OpenMetadata/pull/16631/files#diff-dafe3b7eb01c725b6f8b53a1531cbdcbf52f26d1fccd8d1e7d664288070b3998) to enable filtering a subset of test cases to run.

#
### Type of change:

- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [x] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

- [x] I have added tests around the new logic.
- [x] For connector/ingestion changes: I updated the documentation.

### CODE REVIEW

File: openmetadata-spec/src/main/resources/json/schema/metadataIngestion/testSuitePipeline.json
Comment (TeddyCr) [inline openmetadata-spec/src/main/resources/json/schema/metadataIngestion/testSuitePipeline.json:1]: do we need to add an `isDefault` flag to this schema as well. So we can identify the default ingestion pipeline?

Comment (sushi30) [reply openmetadata-spec/src/main/resources/json/schema/metadataIngestion/testSuitePipeline.json:1]: `null` value resolves the default behavior. its documented in the description. I'd rather avoid adding this field is it has the potential for ambiguity such as:

```yaml
isDefault: true
testCases: ["what_will_this_do"]
```

Comment (TeddyCr) [reply openmetadata-spec/src/main/resources/json/schema/metadataIngestion/testSuitePipeline.json:1]: Sounds good. The behavior will be we can have multiple "default" pipelines right?

Comment (sushi30) [reply openmetadata-spec/src/main/resources/json/schema/metadataIngestion/testSuitePipeline.json:1]: yes. this is the same as current behavior. and as we are doing now we can enforce things on the UI side.

**Review by TeddyCr — APPROVED** [review]

**Pull request merged**